src/DwitTech.AccountService.Core/Services/ActivationService.cs

```diff
@@ -11,40 +11,63 @@ namespace DwitTech.AccountService.Core.Services
 {
     public class ActivationService : IActivationService
     {
-        private readonly IConfiguration _configuration; //Config instance for GetBaseUrl method
+        private readonly IConfiguration _configuration;
         private readonly IUserRepository _userRepository;
+        private readonly IHttpClientFactory _httpClientFactory;
+
+        public ActivationService(IConfiguration configuration, 
+            IUserRepository userRepository, 
+            IHttpClientFactory httpClientFactory)
         private readonly IEmailService _emailService;
         public ActivationService(IConfiguration configuration, IEmailService emailService, IUserRepository userRepository)
         {
             _configuration = configuration;
             _userRepository = userRepository;
+            _httpClientFactory = httpClientFactory;
             _emailService = emailService;
         }
 
-        private string GetActivationCode()
+        private static string GetActivationCode()
         {
             var activationCode = StringUtil.GenerateUniqueCode();
             return activationCode;
         }
 
-        private string GetBaseUrl()
+        private async Task<string> GetActivationUrl(int userId)
```

Comment:
This is an old implementation of this method. The correct one is below. remove this